### ISSUE TITLE
Build Breakage 10.2.0 when using --without-x configure option

### ISSUE THREAD

**Issue by reubenfarrelly** [vmware/open-vm-tools#228]
It appears that there may be a logic problem in the configure script when X is not installed (or rather, not wanted).

This is a regression from 10.1.5, where with the same configure options this build succeeds.

The ./configure I am using is from a Gentoo ebuild, where I am seeing the problem.  However I can reproduce by just extracting the distribution tar file and running with the following configure options:

./configure --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --disable-dependency-tracking --disable-silent-rules --docdir=/usr/share/doc/open-vm-tools-10.2.0 --htmldir=/usr/share/doc/open-vm-tools-10.2.0/html --libdir=/usr/lib64 --disable-deploypkg --disable-static --disable-tests --with-dnet --disable-docs --disable-grabbitmqproxy --disable-vgauth --disable-multimon --without-icu --with-pam --with-pic --without-gtk2 --without-gtkmm --without-gtk3 --without-gtkmm3 --without-x --without-kernel-modules --without-root-privileges --with-kernel-release=4.15.0-gentoo

which leads to:

checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
configure: error: 'need either gtk2 or gtk3 or build with --without-x to disable building desktop plugins'

You can see that gtk2 and gtk3 are disabled, and --without-x is actually there, but seems to be ignored.

Note that this VM (and others I have) do not have X on them, they are command line only and there is no need or requirement for any sort of GUI.

This is with: open-vm-tools-10.2.0-7253323 .

**Comment by oliverkurth** [comment]
I tried to reproduce this myself. So this works:

`./configure --without-x`

But this does not:

```
# ./configure --without-x --without-gtk3 --without-gtk2
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking build system type... (cached) x86_64-unknown-linux-gnu
configure: error: 'need either gtk2 or gtk3 or build with --without-x to disable building desktop plugins'

```

Is there a reason you need to specify `--without-gtk3 --without-gtk2` when using `--without-x`?

**Comment by reubenfarrelly** [comment]
Those --without-gtk arguments are actually being set by the Gentoo ebuild, which appears to be doing this because I have globally disabled -X windows support.
It seems that by removing the --without-gtk3 flag the build then proceeds normally.  But it's pretty unintuitive, as I don't even have gtk3 installed in the first place.

**Comment by oliverkurth** [comment]
Okay, looks like we should just ignore the `--with{,out}-gtk{2,3}` options when `--without-x` is given.

I am working on a fix and will file an internal bug for this.

**Comment by oliverkurth** [comment]
FYI, the fix will be in the stable-10.2.x and devel branches soon.

**Comment by stanguturi** [comment]
This is fixed with cf4f582e5b3c2df25aa775d77d458a15583a05ad and 9af84874a5294d0e9886156a07b539f3c93d2982.
